sectionsUtilityTool/src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionUtilityToolFilter.java

```diff
@@ -0,0 +1,196 @@
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.naming.directory.SearchControls;
+import javax.naming.directory.SearchResult;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import edu.umich.its.lti.utils.PropertiesUtilities;
+
+
+
+public class SectionUtilityToolFilter implements Filter {
+
+	private static Log M_log = LogFactory.getLog(SectionUtilityToolFilter.class);
+	
+	private static final String OU_GROUPS = "ou=Groups";
+	private static final String LDAP_CTX_FACTORY = "com.sun.jndi.ldap.LdapCtxFactory";
+	protected static final String SYSTEM_PROPERTY_FILE_PATH_SECURE = "sectionsToolPropsPathSecure";
+	protected static final String SYSTEM_PROPERTY_FILE_PATH_LESS_SECURE = "sectionsToolPropsPathLessSecure";
+	protected static final String PROPERTY_CANVAS_ADMIN = "canvas.admin.token";
+	protected static final String PROPERTY_CANVAS_URL = "canvas.url";
+	protected static final String PROPERTY_USE_TEST_URL = "use.test.url";
+	protected static final String PROPERTY_LDAP_SERVER_URL = "ldap.server.url";
+	private static final String PROPERTY_AUTH_GROUP = "mcomm.group";
+	private static final String TEST_USER = "testUser";
+	private String providerURL = null;
+	private String mcommunityGroup = null;
+	private boolean isTestUrlEnabled=false;
+	protected static Properties appExtSecurePropertiesFile=null;
+	protected static Properties appExtLessSecurePropertiesFile=null;
+	private static final String FALSE = "false";
+	
+	@Override
+	public void init(FilterConfig filterConfig) throws ServletException {
+		M_log.debug("Filter Init(): Called");
+		getExternalAppProperties();
+		
+	}
+
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response,
+			FilterChain chain) throws IOException, ServletException {
+		M_log.debug("doFilter: Called");
+		HttpServletRequest useRequest = (HttpServletRequest) request;
+		HttpServletResponse useResponse=(HttpServletResponse)response;
+		if(!checkForAuthorization(useRequest)) {
+			useResponse.sendError(403);
+			return;
+		}
+		chain.doFilter(useRequest, response);
+	}
+
+	@Override
+	public void destroy() {
+		M_log.debug("destroy: Called");
+	}
+	
+	protected void getExternalAppProperties() {
+		M_log.debug("getExternalAppProperties(): called");
+		String propertiesFilePathSecure = System.getProperty(SYSTEM_PROPERTY_FILE_PATH_SECURE);
+		String propertiesFilePathLessSecure = System.getProperty(SYSTEM_PROPERTY_FILE_PATH_LESS_SECURE);
+		if (!isEmpty(propertiesFilePathSecure)) {
+			appExtSecurePropertiesFile=PropertiesUtilities.getPropertiesObjectFromURL(propertiesFilePathSecure);
```

Comment:
Generally we use a "secure" properties file to define information that shouldn't be checked into source control -- I don't think this is the case, and don't think there's any benefit to having 2 files that justifies the cost of opening 2 files. Am I missing something?